storage-operator/pkg/apis/storage/v1alpha1/volume_types.go

```diff
@@ -66,6 +148,25 @@ type Volume struct {
 	Status VolumeStatus `json:"status,omitempty"`
 }
 
+// Return the volume size.
+func (self *Volume) GetSize(disk_size resource.Quantity) resource.Quantity {
+	if self.Spec.SparseLoopDevice != nil {
+		return self.Spec.SparseLoopDevice.Size
+	}
+	// RawBlockDevice use the whole disk, so return the disk size as size.
+	return disk_size
+}
+
+// Return the volume path on the node.
+func (self *Volume) GetPath() string {
+	if self.Spec.SparseLoopDevice != nil {
+		return fmt.Sprintf(
+			"/var/lib/metalk8s/storage/sparse/%s", self.UID,
```

Comment:
Will this work? This path will be a file, and the `local` storage provider built into Kubelet doesn't handle those.

Reply:
Good catch!

It should be `/dev/disk/by-id/<self.UID>`

Reply:
Related follow-up: https://kubernetes.slack.com/archives/C09QZFCE5/p1564493300083700